fedimint-core/src/db/mod.rs

```diff
@@ -455,6 +457,9 @@ impl Database {
                     }
                 }
             }
+            let delay = (2u64.pow(curr_attempts.min(8) as u32) * 10).min(2000)
```

Comment:
Not a big deal and I doubt it matters here, but adding `0..30` last is kind of meh. Ideally one wants to randomize for the wide range (as wide as waiting time) to break any possibilities of two things retrying in any kind of a lockstep. So instead of `exp + gen_range(0..30)`, something like `gen_range(exp..(exp * 2))` or `gen_range((exp / 2)..(exp + (exp / 2)))` to keep the expected time the same.

Reply:
done